packages/charts-utils/src/color/generate-continous-color-scale/generate-continous-color-scale.ts

```diff
@@ -0,0 +1,74 @@
+import { scaleLinear } from 'd3-scale';
+import { colors as palette } from '@keen.io/colors';
+
+/**
+ * Generate a linear color scale for heatmap, choropleth and sliders
+ *
+ * @param minValue - minimal data value
+ * @param maxValue - maximal data value
+ * @param steps - amount of steps for scale
+ * @param colors - colors array
+ * @return a color scale
+ *
+ */
+
+const generateContinousColorScale = (
+  minValue: number,
+  maxValue: number,
+  steps = 2,
+  colors: string[]
+) => {
+  if (steps <= 1 || colors.length === 1) {
+    const positiveColor = colors.length === 1 ? 'red' : colors[1];
```

Comment:
`colors.red['500']`?

Reply:
This is that part I mentioned today, to create an 'opposite' color based on `colors[0]` when only one is available in the palette.